### PR TITLE
Add announcement wrap to department radio messages for non-player sources

### PR DESCRIPTION
**Pull request by HoofedEar**
## About the PR 
Before, things like the salvage magnet would "say" that it's pulling in debris to the Salvage radio channel
Now, whenever something that isn't a player (lacks `ActorComponent`, idk what is a better way to check) it gives the message the formatting of a proper announcement

**Screenshots**

Screen Shot 2022-07-24 at 9 11 54 AM

**Changelog**

:cl:
N/A - too minor of tweak

### CODE REVIEW

File: Content.Server/Headset/HeadsetComponent.cs [hunk 1]
```diff
@@ -63,6 +63,18 @@ public void Receive(string message, RadioChannelPrototype channel, EntityUid sou
...
+            if (!_entMan.TryGetComponent(source, out ActorComponent? _))
+            {
+                msg = new MsgChatMessage
+                {
+                    Channel = ChatChannel.Radio,
+                    Message = message,
+                    //Square brackets are added here to avoid issues with escaping
+                    MessageWrap = Loc.GetString("announcement-radio-message-wrap", ("color", channel.Color), ("channel", $"\\[{channel.LocalizedName}\\]"))
+                };
+            }
```
Comment (metalgearsloth) [inline Content.Server/Headset/HeadsetComponent.cs:76]: This is inherently going to break with NPCs.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Server/Headset/HeadsetComponent.cs (hunk 1 above)
Comment (HoofedEar) [reply Content.Server/Headset/HeadsetComponent.cs:76]: Do we have NPCs already? I can add a check to make sure they aren't an NPC either

Comment (HoofedEar) [reply Content.Server/Headset/HeadsetComponent.cs:76]: Alternatively, what is a check I could do that won't break with NPCs?

Comment (metalgearsloth) [reply Content.Server/Headset/HeadsetComponent.cs:76]: Have the caller handle it, or alternatively add a method for it if we expect many more.

Comment (HoofedEar) [reply Content.Server/Headset/HeadsetComponent.cs]: Added changes

File: Content.Server/Headset/HeadsetComponent.cs [hunk 2]
```diff
@@ -63,6 +63,18 @@ public void Receive(string message, RadioChannelPrototype channel, EntityUid sou
...
+            if (announcement)
+            {
+                msg = new MsgChatMessage
+                {
+                    Channel = ChatChannel.Radio,
+                    Message = message,
+                    //Square brackets are added here to avoid issues with escaping
+                    MessageWrap = Loc.GetString("announcement-radio-message-wrap", ("color", channel.Color), ("channel", $"\\[{channel.LocalizedName}\\]"))
+                };
+            }
```
Comment (metalgearsloth) [inline Content.Server/Headset/HeadsetComponent.cs:76]: Should be if else as it creates an unnecessary message.

Comment (HoofedEar) [reply Content.Server/Headset/HeadsetComponent.cs:76]: Could you explain further? Right now, a `MsgChatMessage` is created outside of the if statement and is simply overridden, do you want me to have it create a blank `MsgChatMessage` and have it define the value with if/else?

Comment (metalgearsloth) [reply Content.Server/Headset/HeadsetComponent.cs:76]: The initial message may be redundant and I'm not sure if the compiler would optimise it out so better to just make it an if-else to make the intend clearer.

**Pull request closed without merging**